### DISCUSSION TITLE
Font SVG batch edition

### DISCUSSION THREAD

**Discussion by tobiasBora** [spakin/SimpInkScr#125] [Q&A]
This project looks great, but I can't find how to do batch edition on a SVG Font, like this one (to download it, right click on the white square, and save):

![NimbusSanL-BoldItal-orig](https://github.com/spakin/SimpInkScr/assets/2164118/f89d9148-fae8-45cc-bcc5-7c97461d0b99)

For instance let's say I want to enlarge all glyphs (one can also add a stroke, stroke to path, ungroup and union the stroke and fill, but let's start simple), I can do it manually like this:

![Peek 31-01-2024 11-53](https://github.com/spakin/SimpInkScr/assets/2164118/78cc0600-8264-4c7b-9b48-6fd37cf8b15a)

but it is quite tedious to repeat on all glyphs. Would it be possible to use SimpInkScr to make this easier?

**Comment by spakin** [comment]
> Would it be possible to use SimpInkScr to make this easier?

I believe so.  Iterate over `all_shapes()` to process each glyph in turn.  Then use `translate_path` and `scale_path` to adjust the size and position of the glyph.  See the [Modifying existing objects](https://github.com/spakin/SimpInkScr/wiki/Modifying-existing-objects) wiki page for the relevant documentation.

Path operations like union are documented in [Path operations](https://github.com/spakin/SimpInkScr/wiki/Path-operations).

**Reply by tobiasBora** [reply]
Thanks for your quick answer, unfortunately it is not working. For instance, if I run:
```
for obj in all_shapes():
    obj.rotate(45)
```
then it will do nothing since the glyphs are not yet loaded in a new layer: when you open an SVG font in Inkscape, you need to click on "Edit" for each glyph to move it to their own layer (not to say that if you have hundreds of symbols it can take some time). Then the above code works… but its not enough: one needs to bring back the layer on the glyph by selecting first the shape before doing `Obtain curves`/`Obtenir les courbes` as in my above video. I also can't find how to automate this.

**Reply by tobiasBora** [reply]
Also, is there a way to expand the path (very different from scaling), like in

![image](https://github.com/spakin/SimpInkScr/assets/2164118/d1ade73a-edfc-47cc-a4ae-b5283e6e67e8)

I can't find an action to use like in `apply_action('object-stroke-to-path', shape)`.

**Reply by spakin** [reply]
I can't find an action either.  With English-language menus, {Éroder, Dilater, Décalage dynamique, Décalage lié} are named {Inset, Outset,  Dynamic Offset, Linked Offset}.  I searched for these in Inkscape's action list but didn't find any matching path actions:
```console
$ inkscape --action-list | egrep -i 'inset|outset|offset'
org.inkscape.effect.bluredge:  Inset/Outset Halo
org.inkscape.effect.bluredge.noprefs:  Inset/Outset Halo
org.inkscape.effect.filter.f098:  In and out glow with a possible offset and colorizable flood
org.inkscape.effect.filter.f098.noprefs:  In and out glow with a possible offset and colorizable flood
org.inkscape.effects.inset-shadow:  Inset Shadow
org.inkscape.effects.inset-shadow.noprefs:  Inset Shadow
```

**Comment by spakin** [comment]
> Thanks for your quick answer, unfortunately it is not working. For instance, if I run:
> 
> ```
> for obj in all_shapes():
>     obj.rotate(45)
> ```
> 
> then it will do nothing since the glyphs are not yet loaded in a new layer: when you open an SVG font in Inkscape, you need to click on "Edit" for each glyph to move it to their own layer (not to say that if you have hundreds of symbols it can take some time). Then the above code works… but its not enough: one needs to bring back the layer on the glyph by selecting first the shape before doing `Obtain curves`/`Obtenir les courbes` as in my above video. I also can't find how to automate this.

I didn't realize the glyphs do not appear as paths until they're edited.  I looked through the [`inkex` documentation](https://inkscape.gitlab.io/extensions/documentation/) but failed to find a mechanism to support those operations.  `inkex` defines objects for fonts and glyphs but does not appear to define any methods on those objects that would help with path editing.

I'm afraid you'll have to edit your glyphs one-by-one; sorry.
